src/app/components/wrapper-components/online-game-wrapper/online-game-wrapper.component.ts

```diff
@@ -633,71 +637,79 @@ export class OnlineGameWrapperComponent extends GameWrapper implements OnInit, O
             }
         } else {
-            const user: Player = this.getPlayer();
+            const player: Player = this.observerRole as Player;
             return this.gameService.updateDBBoard(this.currentPartId,
-                                                  user,
+                                                  player,
                                                   encodedMove,
                                                   msToSubstract,
                                                   scores);
         }
     }
     public async resign(): Promise<void> {
+        display(OnlineGameWrapperComponent.VERBOSE, 'OnlineGameWrapperComponent.resign');
         const lastIndex: number = this.getLastIndex();
-        const user: Player = this.getPlayer();
-        const resigner: MinimalUser = this.players[this.observerRole % 2].get();
-        const victoriousOpponent: MinimalUser = this.players[(this.observerRole + 1) % 2].get();
-        await this.gameService.resign(this.currentPartId, lastIndex, user, victoriousOpponent, resigner);
+        const player: Player = this.observerRole as Player;
+        const resigner: MinimalUser = this.players[this.observerRole.value % 2].get();
+        const victoriousOpponent: MinimalUser = this.players[(this.observerRole.value + 1) % 2].get();
+        await this.gameService.resign(this.currentPartId, lastIndex, player, victoriousOpponent, resigner);
     }
     private getLastIndex(): number {
         return this.currentPart.data.lastUpdate.index;
     }
     public async reachedOutOfTime(player: 0 | 1): Promise<void> {
         display(OnlineGameWrapperComponent.VERBOSE, 'OnlineGameWrapperComponent.reachedOutOfTime(' + player + ')');
         const lastIndex: number = this.getLastIndex();
-        const user: Player = this.getPlayer();
+        const currentPlayer: Player = this.observerRole as Player;
         this.stopCountdownsFor(Player.of(player));
-        const opponent: MinimalUser = Utils.getNonNullable(this.opponent);
-        if (player === this.observerRole) {
+        if (player === this.observerRole.value) {
             // the player has run out of time, he'll notify his own defeat by time
-            await this.notifyTimeoutVictory(opponent, user, lastIndex, this.authUser.toMinimalUser());
+            const opponent: MinimalUser = Utils.getNonNullable(this.opponent);
+            await this.notifyTimeoutVictory(opponent, currentPlayer, lastIndex, this.authUser.toMinimalUser());
         } else {
             assert(this.endGame === false, 'time might be better handled in the future');
             if (this.opponentIsOffline()) { // the other player has timed out
-                await this.notifyTimeoutVictory(this.authUser.toMinimalUser(), user, player, opponent);
+                const opponent: MinimalUser = Utils.getNonNullable(this.opponent);
+                await this.notifyTimeoutVictory(this.authUser.toMinimalUser(), currentPlayer, player, opponent);
```

Comment:
le pattern des notifySomekindofVictory qui prend currentPlayer en argument c'p-e un peu débile, c'pas toujours this.authUser.getCurrentUser qu'est dedans ? ça me semble enhancable, t'en sent tu l'âme ?

Reply:
non c'est pas toujours le cas ! Regarde 5 lignes plus haut: en fonction de qui time out, il utilise pas toujours le current user.
Après oui y'a des autres améliorations possibles mais laissons ainsi pour le moment au vu de la taille du ticket :D